### DISCUSSION TITLE
Set active tab programmatically

### DISCUSSION THREAD

**Discussion by ashleyhood** [filamentphp/filament#7463] [Ideas]
I have tabs that are dynamically hidden and visible based on another field. If the tab that is active is hidden then you get a weird state where tabs are visible but no tab is active.

Would it work having a method on the `Tab` called `->active()` that can set which Tab is active programmatically.

**Comment by howdu** [comment]
I did something similar where we needed to add tabs dynamically but used browser events for this scenario.

```blade
    x-on:refresh-tabs.window="tab = $event.detail.tabId || getTabs()[@js($getActiveTab()) - 1]"
```

```php
$livewire->dispatchBrowserEvent('refresh-tabs');

// Or
$livewire->dispatchBrowserEvent('refresh-tabs', ['tabId' => $nextTabUuid]);
```
